packages/helix-shared-process-queue/src/process-queue.js

```diff
@@ -65,6 +136,12 @@ export default async function processQueue(queue, fn, maxConcurrent = 8) {
   }
 
   for await (const value of iter) {
+    if (abortSignal?.aborted) {
+      return results;
+    }
+
+    await waitForToken();
```

Comment:
I would turn this around. so that it doesn't run the handler if it "wakes up" after the waitForToken and is aborted

Reply:
Excellent point, thanks @tripodsan